dm/pkg/parser/common.go

```diff
@@ -108,11 +108,11 @@ func FetchDDLTables(schema string, stmt ast.StmtNode, flavor utils.LowerCaseTabl
 	// special cases: schema related SQLs doesn't have tableName
 	switch v := stmt.(type) {
 	case *ast.AlterDatabaseStmt:
-		return []*filter.Table{genTableName(v.Name, "")}, nil
+		return []*filter.Table{genTableName(v.Name.L, "")}, nil
```

Comment:
we should check `flavor utils.LowerCaseTableNamesFlavor` to decide use .O (origin) or .L (lowercase)

Reply:
previous we use `.O`, see https://github.com/pingcap/tidb/pull/35668/files#diff-157e54ddd07b46a75c54cc2436fdc573d725f91c45e17c4fada5b39e74c86f32L3658

Reply:
I think previous behaviour is not right 😂

Reply:
will left a todo